contrib/completion/bash/docker

```diff
@@ -550,17 +550,18 @@ __docker_complete_nodes() {
 # output to the IDs or names of matching items. This setting takes
 # precedence over the environment setting.
 __docker_services() {
-	local fields='$2'  # default: service name only
-	[ "${DOCKER_COMPLETION_SHOW_SERVICE_IDS}" = yes ] && fields='$1,$2' # ID & name
+	local format='{{.Name}}'  # default: service name only
+	[ "${DOCKER_COMPLETION_SHOW_SERVICE_IDS}" = yes ] && format='{{.ID}} {{.Name}}' # ID & name
 
 	if [ "$1" = "--id" ] ; then
-		fields='$1' # IDs only
+		format='{{.ID}}' # IDs only
 		shift
 	elif [ "$1" = "--name" ] ; then
-		fields='$2' # names only
+		format='{{.Name}}' # names only
 		shift
 	fi
-        __docker_q service ls "$@" | awk "NR>1 {print $fields}"
+
+  __docker_q service ls -q --format "$format" --filter "name=$1"
```

Comment:
Because you dropped the `"$@"` here, the behavior does not mach the documentation any more. Additional arguments are no longer considered but break the function.
I suggest you change this line to
```suggestion
  __docker_q service ls --quiet --format "$format" "$@"
```
and add `--filter "name=$current"` to the function call in `__docker_complete_services`.

This way, the responsibility for filtering services stays in `__docker_complete_services` (where it belongs) and `__docker_services` does not change its semantics.

Reply:
@albers Great feedback. Will get on to doing this ASAP.